tests/diagnostics/ctor-retval-legal.slang

```diff
@@ -0,0 +1,82 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK): -shaderobj -output-using-type
```

Comment:
My understanding is that "diagnostics" directory is for testing the error messages.
I think this file should be relocated to elsewhere because it is not for error test.

Reply:
done